### PR TITLE
systemd::network: Validate if content/source are set for file resource

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
copied it from https://github.com/voxpupuli/puppet-bird/blob/master/manifests/snippet.pp#L25-L30

File: manifests/network.pp [hunk 1]
```diff
@@ -19,6 +19,14 @@
     $notify = undef
   }
 
+  if $ensure == 'file' {
+    if $content =~ Undef and $source =~ Undef {
+      fail('you need to set content or source')
```
Comment (ekohl) [inline manifests/network.pp:24]:
```suggestion
      fail('Either content or source must be set but not both')
```

File: manifests/network.pp [hunk 2]
```diff
@@ -19,6 +19,14 @@
...
+    }
+    if $content =~ NotUndef and $source =~ NotUndef {
+      fail('you can only set $content or $source')
```
Comment (ekohl) [inline manifests/network.pp:27]:
```suggestion
      fail('Either content or source must be set')
```

File: spec/defines/network_spec.rb [hunk 3]
```diff
@@ -61,6 +61,25 @@
 
           it { is_expected.to create_file("/etc/systemd/network/#{title}").that_notifies('Service[systemd-networkd]') }
         end
+        context 'without content and without source' do
+          let(:title) { 'wg0.netdev' }
```
Comment (ekohl) [inline spec/defines/network_spec.rb:65]: There is already an inherited title, right?
```suggestion
```

File: spec/defines/network_spec.rb [hunk 4]
```diff
@@ -61,6 +61,25 @@
...
+          let :params do
+            {}
+          end
+
+          it { is_expected.to compile.and_raise_error(%r{you need to set}) }
+        end
+        context 'with content and source' do
+          let(:title) { 'wg0.netdev' }
```
Comment (ekohl) [inline spec/defines/network_spec.rb:73]: Same here
```suggestion
```

**Pull request merged**